plugins/LayerView/LayerView.qml

```diff
@@ -292,7 +303,7 @@ Item
                         color: UM.Theme.getColor(model.colorId)
                         border.width: UM.Theme.getSize("default_lining").width
                         border.color: UM.Theme.getColor("lining")
-                        visible: view_settings.show_legend
+                        visible: (UM.Preferences.getValue("layerview/layer_view_type") != 0) && view_settings.show_legend
```

Comment:
This is now exactly the same as the parent, so this line has no effect any more. If the parent label is invisible, this rectangle will be too.